django/contrib/admindocs/views.py

```diff
@@ -411,8 +412,7 @@ def simplify_regex(pattern):
     """
     pattern = replace_named_groups(pattern)
     pattern = replace_unnamed_groups(pattern)
-    # clean up any outstanding regex-y characters.
-    pattern = pattern.replace('^', '').replace('$', '').replace('?', '')
+    pattern = re.sub(r"(\\[bBAZ]|[\^\$\?])", "", pattern)
```

Comment:
We can avoid the backslashes in `[…]`, even for `^` as long as it isn't first to avoid turning it into an inverse character set (`[^…]`):
```suggestion
    pattern = re.sub(r"(\\[bBAZ]|[?^$])", "", pattern)
```